dissect/target/plugins/os/unix/log/messages.py

```diff
@@ -48,68 +42,65 @@ def messages(self):
-                    # FIXME watch for change of year in file!!!
-                    file_mtime = self.target.fs.get(str(f)).stat().st_mtime
-                    year = datetime.datetime.fromtimestamp(file_mtime).year
-
-                    fh = f.open()
-                    if f.name[-2:] == "gz":
-                        fh = zlib.decompress(fh.read(), 31)
-                        try:
-                            fh = StringIO(fh.decode())
-                        except UnicodeDecodeError:
-                            pass
-                    if "bz2" in f.name:
-                        fh = bz2.decompress(fh)
-                        fh = StringIO(fh.decode())
-
-                    for line in fh:
-
-                        # Line can be a byte object on debian and centos
-                        if isinstance(line, bytes):
-                            line = line.decode("utf-8").strip()
-                        else:
-                            line = line.strip()
-
-                        if not line:
-                            continue
-
-                        try:
-                            ts = datetime.datetime.strptime(RE_TS.search(line).groups()[0], "%b %d %H:%M:%S")
-                            ts = ts.replace(year=year)
-                        except Exception:
-                            # Set ts to epoch 1970-01-01 if we could not convert
-                            ts = datetime.datetime.utcfromtimestamp(0)
-
-                        try:
-                            daemon = RE_DAEMON.search(line).groups()[0]
-                        except Exception:
-                            daemon = None
-
-                        try:
-                            pid = RE_PID.search(line)
-                            if pid is not None:
-                                pid = pid.groups()[0]
-                        except Exception:
-                            pid = None
-
-                        try:
-                            msg = RE_MSG.search(line).groups()[0]
-                            if msg is None:
-                                msg = line
-                        except Exception:
-                            msg = line
-
-                        yield MessagesRecord(
-                            ts=ts,
-                            filepath=f.path,
-                            daemon=daemon,
-                            pid=pid,
-                            msg=msg,
-                            raw=line,
-                            _target=self.target,
-                        )
+
+        log_files: [TargetPath] = list(self.target.fs.path("/var/log/").glob("syslog*")) + list(
+            self.target.fs.path("/var/log/").glob("messages*")
+        )
+
+        for log_file in log_files:
+
+            file_ctime = self.target.fs.get(str(log_file)).stat().st_ctime
+            year_file_created = datetime.fromtimestamp(file_ctime).year
```

Comment:
Preferably use `ts.from_unix` from `dissect.util`, since that will make sure it's using the UTC timezone and not a local one.